fsspec/implementations/http.py

```diff
@@ -262,9 +263,19 @@ async def _put_file(
         **kwargs,
     ):
         async def gen_chunks():
-            with open(rpath, "rb") as f:
-                callback.set_size(f.seek(0, 2))
-                f.seek(0)
+            # Support passing arbitrary file-like objects
+            # and use them instead of streams.
+            if isinstance(rpath, io.IOBase):
```

Comment:
I *think* all of our own file-like instances pass this, but we could consider testing if the path is a string instead.

Reply:
then we would give away PathLike objects, we can test it against `isinstance(rpath, os.PathLike)` but I am not sure if that would break compatibility in any way, so this is seemed more right. But if we decide to formalize what can be passed to `get_file`/`put_file` (adding streams as an official fsspec feature for example), then we can offer some sort of utility to either open the path, or return the stream directly and do the checks locally with against these defined standards.

Reply:
Good point